interface/modules/custom_modules/oe-module-weno/scripts/file_download.php

```diff
@@ -105,87 +105,98 @@ function download_zipfile($fileUrl, $zipped_file)
+                    $address_line_2 = $line[8];
+                    $city = $line[9];
+                    $state = $line[10];
+                    $zipcode = str_replace(['[', ']'], '', $line[11]);
+                    $country = $line[12];
+                    $international = $line[13];
+                    $pharmacy_phone = str_replace(['[', ']'], '', $line[16]);
+                    $on_weno = $line[21];
+                    $test_pharmacy = $line[17];
+                    $state_wide_mail = $line[18];
+                    $fullDay = $line[22];
+                } else {
+                    $ncpdp = str_replace(['[', ']'], '', $line[3]);
+                    $npi = str_replace(['[', ']'], '', $line[7]);
+                    $business_name = $line[8];
+                    $city = $line[11];
+                    $state = $line[12];
+                    $zipcode = str_replace(['[', ']'], '', $line[14]);
+                    $country = $line[15];
+                    $address_line_1 = $line[9];
+                    $address_line_2 = $line[10];
+                    $international = $line[16];
+                    $pharmacy_phone = str_replace(['[', ']'], '', $line[20]);
+                    $county = $line[33];
+                    $on_weno = $line[37];
+                    $compounding = $line[41];
+                    $medicaid_id = $line[45];
+                    $dea = $line[44];
+                    $test_pharmacy = $line[29];
+                    $fullDay = $line[40];
+                    $state_wide_mail = $line[47];
+                }
+
+                $insertdata['ncpdp'] = $ncpdp;
+                $insertdata['npi'] = $npi;
+                $insertdata['business_name'] = $business_name;
+                $insertdata['address_line_1'] = $address_line_1;
+                $insertdata['address_line_2'] = $address_line_2;
+                $insertdata['city'] = $city;
+                $insertdata['state'] = $state;
+                $insertdata['zipcode'] = $zipcode;
+                $insertdata['country'] = $country;
+                $insertdata['international'] = $international;
+                $insertdata['pharmacy_phone'] = $pharmacy_phone;
+                $insertdata['on_weno'] = $on_weno;
+                $insertdata['test_pharmacy'] = $test_pharmacy;
+                $insertdata['state_wide_mail'] = $state_wide_mail;
+                $insertdata['fullDay'] = $fullDay;
+
+                if ($data['Daily'] == 'Y') {
+                    $insertPharmacy->updatePharmacies($insertdata);
+                } else {
+                    $insertPharmacy->insertPharmacies($insertdata);
+                }
+
+                ++$l;
             }
-
-            ++$l;
         }
+    } catch (Exception $e) {
+        throw new RuntimeException($e->getMessage(), $e->getCode(), $e);
     }
     $wenolog->insertWenoLog("pharmacy", "Success");
     fclose($records);
+    sqlStatementNoLog('COMMIT');
+    sqlStatementNoLog('SET autocommit=1');
```

Comment:
I wonder if I should be concerned with default settings for PHP and engine resources with approx 100, 000 inserts?
Perhaps safer to batch maybe every 10,000 or so!